cmd/arcaflow/main.go

```diff
@@ -57,53 +58,42 @@ func main() {
 	workflowFile := "workflow.yaml"
 	printVersion := false
 
-	flag.BoolVar(&printVersion, "version", printVersion, "Print Arcaflow Engine version and exit.")
-	flag.StringVar(
-		&configFile,
-		"config",
-		configFile,
-		"The Arcaflow configuration file to load, if any.",
-	)
-	flag.StringVar(
-		&input,
-		"input",
-		input,
-		"The workflow input file to load. May be outside the workflow directory. If no input file is passed, "+
-			"the workflow is assumed to take no input.",
-	)
-	flag.StringVar(
-		&dir,
-		"context",
-		dir,
-		"The workflow directory to run from. Defaults to the current directory.",
-	)
-	flag.StringVar(
-		&workflowFile,
-		"workflow",
-		workflowFile,
-		"The workflow file in the current directory to load. Defaults to workflow.yaml.",
+	const (
+		helpUsage    = "Display this help output."
+		versionUsage = "Print Arcaflow Engine version and exit."
+		configUsage  = "The Arcaflow configuration file to load, if any."
+		inputUsage   = `The workflow input file to load. May be outside
+                           the workflow directory. If no input file is
+                           passed, the workflow is assumed to take no input.`
+		contextUsage = `The workflow directory to run from. Defaults 
+                           to the current directory.`
+		workflowUsage = `The workflow file in the current directory to load.
+                           Defaults to workflow.yaml.`
 	)
+	flag.BoolVar(&printVersion, "version", printVersion, versionUsage)
+	flag.BoolVar(&printVersion, "v", printVersion, versionUsage+" (shorthand)")
```

Comment:
I don't really like the "shorthand", I don't think that information is necessary or particularly relevant, and the linter doesn't like the repeated literal. I'd just remove it entirely. There are two options to get version, or help, and that's a pretty universal concept I doubt needs any explanation. (The fact that you claim the Go package will accept either `-version` or `--version` is ugly since traditionally the old-style single character flags can be combined like `-vh` which is at best really confusing if it also allows `-version`, but I guess that's a completely separate ugliness...)

Reply:
I'm not entirely happy with how the `flag` plugin works, honestly. I wonder if there is an alternative that uses more normal conventions natively.

Reply:
It seems that `flag` is the available built-in package and the alternatives I can fine are all third-party.

Reply:
Apparently, `flag` is Google's conceit that "the era of one-letter options has passed" (i.e., their universal repo doesn't use them) and therefore requiring "--" just wastes space. And while that argument might have been perfectly acceptable for their own internal code, releasing it as the default built-in package seems rather inconsiderately arrogant. (Then again, this is Google.)

I'm not going to make a strong argument for choosing an alternative and more gnu culturally compatible parsing package in this context, but such an argument could be made; and if we're going to support short options like `-c` and `-h` and `-v`, we **ought** to be able to type `-chv config-file` and `flag` doesn't do that, making single letter options, culturally speaking, "wrong".

But whatever ... I won't throw a grenade on the PR for this beyond reiterating that adding single-character flags with the `flags` package is ridiculous. 😏